pkg/query-service/app/http_handler.go

```diff
@@ -302,18 +303,84 @@ func (aH *APIHandler) metricAutocompleteTagValue(w http.ResponseWriter, r *http.
 func (aH *APIHandler) queryRangeMetricsV2(w http.ResponseWriter, r *http.Request) {
 	metricsQueryRangeParams, apiErrorObj := parser.ParseMetricQueryRangeParams(r)
 
-	fmt.Println(metricsQueryRangeParams)
-
 	if apiErrorObj != nil {
 		zap.S().Errorf(apiErrorObj.Err.Error())
 		aH.respondError(w, apiErrorObj, nil)
 		return
 	}
-	response_data := &model.QueryDataV2{
-		ResultType: "matrix",
-		Result:     nil,
+
+	// build queries
+	queries, err := metricsQueryRangeParams.BuildQuery("time_series")
+	if err != nil {
+		aH.respondError(w, &model.ApiError{Typ: model.ErrorBadData, Err: err}, nil)
 	}
-	aH.respond(w, response_data)
+
+	var formulaSubQuery string
+	var results []*[]model.MetricResult
+	// read the result for individual metric queries and build query for formula
```

Comment:
@srikanthccv can we build the entire result in one function, so I can re-use it in alerts. http handler methods will not be callable from alerting module. 

Ideally (reader *clickhouseReader) BuildMetricQueryResult() (result interface{}, err error) or something that I could call from alerts will be really useful

Reply:
Makes sense.